src/lib.rs

```diff
@@ -138,13 +138,13 @@ impl IWICBitmapDecoder_Impl for JXLWICBitmapDecoder {
         );
         // TODO: Proper color context
         unsafe {
-            if !ppicolorcontexts.is_null() && ccount == 1 {
-                ppicolorcontexts
-                    .as_mut()
-                    .unwrap()
-                    .as_mut()
-                    .expect("There should be a color context here")
-                    .InitializeFromMemory(&decoded.icc[..])?;
+            if let Some(context) = ppicolorcontexts.as_mut() {
+                if ccount == 1 {
```

Comment:
Maybe this can even do tuple match? 👀

Reply:
I tried matching and if let a tuple but I found them less readable than just nesting if. Sadly chaining if let is not stable yet.

```suggestion -1
            if let (Some(context), 1) = (ppicolorcontexts.as_mut(), ccount) {
```